### PR TITLE
Added visualisations to the dashboard

### PR DESCRIPTION
**Pull request by IshikaM19**
I have created the dashboard :). 
I've added 2 separate script: The functions_for_database.py has all the functions that are used in dashboard.py. 
I've tried to create a bunch of different visualisations but I'm not sure If I have enough or if some of them are even relevant so please have a look and let me know. I also added all requirements to the requirements.txt :)

### CODE REVIEW

File: Dashboard/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+if __name__ == "__main__":
+
+    st.set_page_config(page_title="LMNH Plant Analysis",
+                       page_icon=":bar_char:",
+                       layout="wide")
+
+    st.title(
+        ":bar_chart: :potted_plant: LMNH Plant Analysis :potted_plant: :bar_chart:")
+
+    load_dotenv()
```
Comment (clv1) [inline Dashboard/dashboard.py:18]: This might not be needed here - it looks like you're already calling load_dotenv() within the get_db_connection() function already.

File: Dashboard/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    st.sidebar.write(plants_data[['plant_id', 'name']].drop_duplicates())
+    st.sidebar.header("Please Filter Here:")
+
+    plant_id = st.sidebar.multiselect(
+        "Select a Plant by it's ID 🪴 :",
+        options=get_unique_plant_ids(plants_data).plant_id
+    )
+    add_selectbox = st.sidebar.selectbox(
+        "Select this box if you would like to view data for the past 24 hours?",
+        ("All data", "Last 24 Hours")
```
Comment (clv1) [inline Dashboard/dashboard.py:39]: Why the question mark? Maybe rephrase along the lines of "Select this box to view data for the past 24 hours:".

File: Dashboard/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if add_selectbox == "All data":
+        st.altair_chart(make_temperature_graph(
+            plants_data, plant_id), use_container_width=True)
+
+    if add_selectbox == "Last 24 Hours":
+        st.altair_chart(make_temperature_graph(
+            plants_data_24, plant_id), use_container_width=True)
+
+    # MOISTURE OF PLANTS AS TIME HAS GONE
+    st.subheader('Moisture overtime', divider='rainbow')
```
Comment (clv1) [inline Dashboard/dashboard.py:63]: "Moisture level over time"

File: Dashboard/dashboard.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    with col2:
+        st.subheader(
+            "How many have been watered per hour in the last 24 hours?")
+
+        st.altair_chart(make_watered_per_hour_chart(plants_data_24))
+
+    # COUNTRIES AND PLANTS
+
+    st.subheader('The Diveristy of the plants based on Country')
```
Comment (clv1) [inline Dashboard/dashboard.py:90]: "Diversity"

File: Dashboard/dashboard_functions.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    query = """SELECT s_alpha.recording_event.*, s_alpha.plant.name, s_alpha.plant.image_url,
+                s_alpha.botanist.first_name, s_alpha.botanist.last_name,
+                s_alpha.origin_location.country
+                FROM s_alpha.recording_event
+                JOIN s_alpha.plant
+                ON s_alpha.plant.plant_id = s_alpha.recording_event.plant_id
+                JOIN s_alpha.botanist
+                ON s_alpha.botanist.botanist_id = s_alpha.recording_event.botanist_id
+                JOIN s_alpha.origin_location
+                ON s_alpha.origin_location.origin_location_id = s_alpha.plant.origin_location_id
```
Comment (clv1) [inline Dashboard/dashboard_functions.py:56]: Great point raised about the JOINs and nice implementation.

File: Dashboard/dashboard_functions.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                WHERE recording_taken > ?;"""
+    data = pd.read_sql(query, conn, params=(LAST_24_LIMIT,))
+    return data
+
+
+def get_unique_plant_ids(plants_data: pd.DataFrame) -> pd.DataFrame:
+    """Returns a dataframe of all the unique plant_id's in ascedning order"""
+
```
Comment (clv1) [inline Dashboard/dashboard_functions.py:64]: spelling.

File: Dashboard/dashboard_functions.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        subset=['botanist_name', 'plant_id'])
+
+    botanists_and_plants = botanists_and_plants.groupby(
+        'botanist_name')['plant_id'].agg(list).reset_index()
+    botanists_and_plants.columns = ['botanist', 'plants_worked_with']
+
+    return botanists_and_plants
+
+
+def make_temperature_graph(plants_data: pd.DataFrame, selected: list) -> alt.Chart:
```
Comment (clv1) [inline Dashboard/dashboard_functions.py:89]: Nice idea for a graph.

**Review by clv1 — CHANGES_REQUESTED** [review]
Changes requested are mostly just minor typos/spelling checks.
One thing worth discussing in our next huddle would be going over would be how we are implementing 24hrs/overall data.
Overall, dashboard code looks great and a really nice selection of graphs added!

**Review by BenBeard115 — APPROVED** [review]
Perfect!
